src/main/java/org/folio/circulation/resources/CheckOutByBarcodeResource.java

```diff
@@ -214,4 +216,9 @@ private void copyOrDefaultLoanDate(JsonObject request, JsonObject loan) {
     return Result.combine(loanResult, inventoryRecordsResult,
       LoanAndRelatedRecords::withItem);
   }
+  private CompletableFuture<Result<LoanAndRelatedRecords>> addItemLocation(
+    LoanAndRelatedRecords relatedRecords) {
+
+    return completedFuture(succeeded(relatedRecords.withItemEffectiveLocationIdAtCheckOut()));
```

Comment:
Personally, I would leave the logic for this change here, rather than in the `LoanAndRelatedRecords` class

Reply:
From this line, I can retrieve the item location ID
```
String itemLocationId = relatedRecords.getLoan().getItem().getLocationId();
```

but I cannot `put` to the loan JSON object from here without calling a setter method on the `LoanAndRelatedRecords`.

Reply:
Sure, I'm suggesting that the existing methods can be used for `LoanAndRelatedRecords` rather than introduce a very specialised method there for this purpose alone